programs/bpf_loader/src/lib.rs

```diff
@@ -281,7 +281,7 @@ macro_rules! create_vm {
             >::zero_filled(stack_size);
             let mut heap = solana_rbpf::aligned_memory::AlignedMemory::<
                 { solana_rbpf::ebpf::HOST_ALIGN },
-            >::zero_filled(heap_size);
+            >::zero_filled(usize::try_from(heap_size).unwrap());
```

Comment:
this `unwrap()` _may_ fail IF target is 16-bit and `heap_size` is too large for that. In that case, panic is proper imo